### PR TITLE
Check for unauthed events via send_join

### PR DESCRIPTION
**Pull request by richvdh**
Based on #131

### CODE REVIEW

**Comment by richvdh** [comment]
This will fail until https://github.com/matrix-org/synapse/pull/10243 lands.

**Review by erikjohnston — APPROVED** [review]

File: tests/federation_room_join_test.go [hunk 1]
```diff
@@ -192,3 +195,70 @@ func TestJoinFederatedRoomWithUnverifiableEvents(t *testing.T) {
...
+		Sender:   alice.UserID,
+		StateKey: &charlie,
+		Content: map[string]interface{}{
+			"membership": "ban",
+		},
+	})
+
+	// charlie sends a make_join for a different user
+	makeJoinResp, err := fedClient.MakeJoin(context.Background(), "hs1", roomID, srv.UserID("charlie2"), federation.SupportedRoomVersions())
+	must.NotError(t, "MakeJoin", err)
```
Comment (kegsay) [inline tests/federation_room_join_test.go:230]: So Dendrite will fail the test here with `M_FORBIDDEN: eventauth: "@charlie2:host.docker.internal" is not allowed to change their membership from "leave" to "join"` which feels like reasonable behaviour. Annoyingly, Complement CI seems to not be running Dendrite tests through to completion causing this not to be picked up.

Currently Complement tests now fail on the Dendrite repo as a result.

Comment (richvdh) [reply tests/federation_room_join_test.go:230]: I think this should work? it should be a public room so `make_join` should work?

Comment (kegsay) [reply tests/federation_room_join_test.go:230]: The root cause here was that `/createRoom` with `{}` defaults to private rooms as per the spec:

> A public visibility indicates that the room will be shown in the published room list. A private visibility will hide the room from the published room list. Rooms default to private visibility if this key is not included. 

But synapse defaults to public it seems.

Comment (richvdh) [reply tests/federation_room_join_test.go:230]: fixed by #158

**Pull request merged**